### PR TITLE
selenium4: migrate to new core

### PR DESCRIPTION
**Pull request by kyryloonufriiev**
I already сreated all the necessary files to specialize abstractions from the `core`. But it still should wait until `core` will be updated to support async `specs.extractId(el)`.

### CODE REVIEW

File: packages/eyes-selenium/src/selenium4/SpecWrappedElement.js [hunk 1]
```diff
@@ -30,7 +30,8 @@ module.exports = {
     return (
       selector instanceof By ||
       TypeUtils.has(selector, ['using', 'value']) ||
-      Object.keys(selector).some(key => key in By)
+      Object.keys(selector).some(key => key in By) ||
```
Comment (amitzur) [inline packages/eyes-selenium/src/selenium4/SpecWrappedElement.js:33]: @kyryloonufriiev function returns true if there's at least one key in `selector` that is also in `By`? How is this logic correct?

File: packages/eyes-selenium/test/it/VisualGridResources.spec.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ describe('VisualGrid Resources', () => {
 
     await eyes.check('region', Target.region(new Region(200, 200, 50, 100)))
 
-    await eyes.check('selector', Target.region('#scroll1'))
+    await eyes.check('selector', Target.region({css: '#scroll1'}))
```
Comment (amitzur) [inline packages/eyes-selenium/test/it/VisualGridResources.spec.js:57]: @kyryloonufriiev wait, the API used to support getting a string and now it doesn't?

File: packages/eyes-selenium/src/selenium4/SpecWrappedElement.js (hunk 1 above)
Comment (kyryloonufriiev) [reply packages/eyes-selenium/src/selenium4/SpecWrappedElement.js:33]: You can see it in the Selenium docs. They support objects like this `{css: '.some'}`

File: packages/eyes-selenium/test/it/VisualGridResources.spec.js (hunk 2 above)
Comment (kyryloonufriiev) [reply packages/eyes-selenium/test/it/VisualGridResources.spec.js:57]: Only Selenium4. I preserve strings support on the Selenium4, but I fix the test to work also with Selenium3.

**Pull request merged**